src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -931,4 +938,62 @@ public Response getHasBeenDeleted(@Context ContainerRequestContext crc, @PathPar
             return ok(dataFileServiceBean.hasBeenDeleted(dataFile));
         }, getRequestUser(crc));
     }
+
+    /**
+     * @param fileIdOrPersistentId Database ID or PID of the data file.
+     * @param dsVersionString The version of the dataset, such as 1.0, :draft,
+     * :latest-published, etc.
+     */
+    @GET
+    @AuthRequired
+    @Path("{id}/versions/{dsVersionString}/citation")
+    public Response getFileCitationByVersion(@Context ContainerRequestContext crc, @PathParam("id") String fileIdOrPersistentId, @PathParam("dsVersionString") String dsVersionString) {
+        try {
+            DataverseRequest req = createDataverseRequest(getRequestUser(crc));
+            final DataFile df = execCommand(new GetDataFileCommand(req, findDataFileOrDie(fileIdOrPersistentId)));
+            Dataset ds = df.getOwner();
+            // Adapted from getDatasetVersionOrDie
+            DatasetVersion dsv = execCommand(handleVersion(dsVersionString, new Datasets.DsVersionHandler<Command<DatasetVersion>>() {
```

Comment:
I would extract common behavior with `getDatasetVersionOrDie` to a common point so we avoid this duplication with getDatasetVersionOrDie.

Furthermode, invoking the static "handleVersion" method,  which belongs to the Datasets class, from the Files class, seems to lack cohesion in my opinion.

I suggest to refactor the common logic by centralizing it in the parent component (AbstractApiBean). This minimizes code duplication, enhances maintainability, and aligns with DRY principles.

Reply:
I was wondering about extracting this code to a common location. I'll give it a try. Thanks.

Reply:
I took a quick swing at refactoring here: https://github.com/pdurbin/dataverse/commit/30b5fc6b544cc0bfde1cb435b3cd8f28b1e28236

However, it wouldn't deploy to I need to look into it more closely.

I'll circle back to this.

Reply:
Address in this commit: 4542b21310 refactor handleVersion, add includeDeaccession param